src/web/core/src/main/java/org/geoserver/web/data/resource/BasicResourceConfig.java

```diff
@@ -457,4 +326,130 @@ public void validate(final Form<?> form) {
             }
         }
     }
+
+    public boolean addOtherSRS(ResourceInfo resourceInfo) {
+
+        // first check if its WFS-NG or WMSStore
+        List<String> otherSRS = getOtherSRSFromWFSOrWMS(resourceInfo);
+
+        if (otherSRS != null) {
+            if (!otherSRS.isEmpty()) {
+                resourceInfo
+                        .getMetadata()
+                        .put(FeatureTypeInfo.OTHER_SRS, String.join(",", otherSRS));
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private String getActualNativeSRSCode(ResourceInfo resourceInfo) {
+
+        try {
+            if (resourceInfo instanceof FeatureTypeInfo) {
+                FeatureTypeInfo ft = (FeatureTypeInfo) resourceInfo;
+                return "EPSG:"
+                        + CRS.lookupEpsgCode(
+                                ft.getFeatureSource(null, null).getInfo().getCRS(), false);
+            } else {
+                CatalogBuilder cb = new CatalogBuilder(GeoServerApplication.get().getCatalog());
+                cb.setStore(resourceInfo.getStore());
+                ReferencedEnvelope bounds = cb.getNativeBounds(resourceInfo);
```

Comment:
What an weird way of get the native bounds, and then using it to retrieve the CRS, is there not a more straightforward way?

Reply:
that is a good point
the function `getActualNativeSRSCode` purpose is to find out if currently selected native SRS is the actual native SRS or user has modified it? That is why I used the CatalogBuilder to get ReferencedEnvelope and then use its Native SRS, I would be glad to modify this, if their exists a simpler way.

Reply:
There is nothing straightforward, each store has its own native fast way. Coverage readers expose the native SRS as a dedicated method in the reader class, cascaded sources have a different approach. One option would be to add a new method in the catalog builder that does the native SRS computation without computing the BBOX, mimicking what getNativeBounds does, but without triggering a full bounds calculation, which as we know, can be really expensive. Then you can do without the if here.